File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        int max=-1;
        for(int i=0;i<nums.length;i++){
            for(int j=i+1;j<nums.length;j++){
                if(nums[i]<nums[j]){
                int temp=nums[j]-nums[i];
                if(temp>max){
                    max=temp;
                }
            }
            }

            }
            return max;
        }
    
        
    }

File: 2144-maximum-difference-between-increasing-elements/2144-maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        minVal = nums[0]
        maxDiff = -1
        for num in nums:
            if num > minVal:
                diff = num - minVal
                maxDiff = max(maxDiff, diff)
            else:
                minVal = num
        return maxDiff
        